resharper/resharper-unity/src/Yaml/Psi/Caches/UnityEditorPropertyValues/SceneHierarchy.cs

```diff
@@ -0,0 +1,268 @@
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using JetBrains.Annotations;
+using JetBrains.Application.PersistentMap;
+using JetBrains.Collections;
+using JetBrains.Diagnostics;
+using JetBrains.Extension;
+using JetBrains.ReSharper.Plugins.Yaml.Psi;
+using JetBrains.ReSharper.Plugins.Yaml.Psi.Parsing;
+using JetBrains.ReSharper.Plugins.Yaml.Psi.Tree;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Files;
+using JetBrains.ReSharper.Psi.Parsing;
+using JetBrains.ReSharper.Psi.Web.WebConfig;
+using JetBrains.Serialization;
+using JetBrains.Text;
+using JetBrains.Util;
+using JetBrains.Util.Collections;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Yaml.Psi.Caches.UnityEditorPropertyValues
+{
+    public class SceneHierarchy
```

Comment:
almost everything is public here, incl public static APIs. not requirements for the threading. 
If this is a free threaded code - these dictionaries won't work. If it is a UI thread only code - mark it please with assertion.